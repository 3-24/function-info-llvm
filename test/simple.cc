#include<stdio.h>

void foo(int * a) {
    *a = 20;
    return;
}

int goo(int a, int b){
    return a+b;
}


int main() {
   int a = 30;
   int b = 0;

    for (a = 0 ; a < 20 ; a++) {
        b += a;
    }

    foo(&a);

    return 0;
}