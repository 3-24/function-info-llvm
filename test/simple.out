[
    {"filename": "/function-info-llvm/test/simple.cc", "name": "foo(int*)", "type": "i32* -> void", "start_line": 3, "end_line": 5},
    {"filename": "/function-info-llvm/test/simple.cc", "name": "goo(int, int)", "type": "i32 -> i32 -> i32", "start_line": 8, "end_line": 9},
    {"filename": "/function-info-llvm/test/simple.cc", "name": "main", "type": "i32", "start_line": 14, "end_line": 23},
]
